### ISSUE TITLE
M3-PPT Export - Endless Export dialog if user don't pause playing from Time controller

### ISSUE THREAD

**Issue by helenayele** [debrief/debrief#3232]
During PPT Export,  if a user does pause recording before it reaches end time of playing from the time controller , we don't have any issues, but if the time player from time controller reaches its end time and it stops playing by itself, then export dialog will be shown continuously/ no end.
Steps to reproduce:
1. Open sample.dpf plot.
2. Zoom in on the area of interest.  The end of sample.dpf is most interesting, so zoom in on that.
3. Focus on period of interest. Move the time slider so that the vessel markers are visible
4. Adjust time stepper priorities:
5. Click on the Edit Time Controller Properties button, in the Time Controller toolbar
       a. Set Small Step to 1 Minute, and Step Interval to ⅕ second.
6. Record engagement
7. Assign Export to PPT by clicking on the drop-down menu next to the play button, and selecting it
8. Press Play button
9. Let events unfold….
10. Don't press pause button, let it ends by itself 
As soon as it stops playing an export dialog is shown, even if you select export it will be shown endlessly and it is not possible to pause recording once this dialog is shown. Even if you can  close this dialog by pressing Cancel. 
Unable to pause the recording until, I terminate the application(Debrief)

Video : -https://gyazo.com/9fe05b013c76cfd78b05d47f92cba5a1
SS : -https://gyazo.com/7978ad0b7fcf483e75f75e7d94b4d1e5- Recording is still on
         https://gyazo.com/0b61d1a2b8c4485130203e48345a86ae

[SWTException_invalid_thread_Access.txt](https://github.com/debrief/debrief/files/2455432/SWTException_invalid_thread_Access.txt)

Expected Result 

Export dialog should be shown only once after it reaches end of recording, and it shouldn't not be shown again either a user exports the current recording or user presses cancel button of export dialog.

**Comment by helenayele** [comment]
In today's test, when the player reaches at the end time, it will not pop up the export dialog instead NullPointerException is thrown - java.lang.NullPointerException
	at org.mwc.cmap.TimeController.recorders.CoordinateRecorder.tidyString(CoordinateRecorder.java:127)
	at org.mwc.cmap.TimeController.recorders.CoordinateRecorder.access$1(CoordinateRecorder.java:125)
	at org.mwc.cmap.TimeController.recorders.CoordinateRecorder$1.run(CoordinateRecorder.java:91)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:185)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:3933)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3564)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1173)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1062)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:156)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:628)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:563)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:151)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:155)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:199)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:137)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:107)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:391)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:246)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:659)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:595)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1501)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1474)

**Comment by helenayele** [comment]
And sometimes I got this logged instead of "Success in export" message. It seems that the export dialog file name has issues - https://gyazo.com/a0cd2b1a9136d0759fc339131d2981b7

java.nio.file.InvalidPathException: Illegal char <:> at index 72: C:\Users\helena\Downloads\debrief_video_Export\DebriefExport-95\12\12 11:26_27.PPTX
	at sun.nio.fs.WindowsPathParser.normalize(Unknown Source)
	at sun.nio.fs.WindowsPathParser.parse(Unknown Source)
	at sun.nio.fs.WindowsPathParser.parse(Unknown Source)
	at sun.nio.fs.WindowsPath.parse(Unknown Source)
	at sun.nio.fs.WindowsFileSystem.getPath(Unknown Source)
	at java.io.File.toPath(Unknown Source)
	at Debrief.ReaderWriter.powerPoint.PackPresentation.pack(PackPresentation.java:63)
	at Debrief.ReaderWriter.powerPoint.PlotTracks.createPptxFromTrackData(PlotTracks.java:678)
	at Debrief.ReaderWriter.powerPoint.PlotTracks.export(PlotTracks.java:865)
	at Debrief.GUI.Views.CoreCoordinateRecorder.exportFile(CoreCoordinateRecorder.java:175)
	at Debrief.GUI.Views.CoreCoordinateRecorder.stopStepping(CoreCoordinateRecorder.java:285)
	at org.mwc.cmap.TimeController.views.TimeController.stopRecording(TimeController.java:3137)
	at org.mwc.cmap.TimeController.views.TimeController.access$16(TimeController.java:3118)
	at org.mwc.cmap.TimeController.views.TimeController$RecordButtonListener.widgetSelected(TimeController.java:290)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:252)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4145)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1055)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:3958)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3561)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1173)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1062)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:156)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:628)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:563)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:151)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:155)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:199)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:137)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:107)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:391)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:246)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:659)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:595)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1501)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1474)

**Comment by IanMayo** [comment]
This should have been fixed in the PR 5 hours ago:
https://github.com/debrief/debrief/pull/3251

**Comment by helenayele** [comment]
Yes, the above 2 comments are based on the new PR. I did git pull and tested it
